Source/Extensions/Blazorise.TreeView/TreeView.razor.cs

```diff
@@ -46,6 +46,13 @@ public override async Task SetParametersAsync( ParameterView parameters )
             treeViewState = treeViewState with { SelectedNodes = paramSelectedNodes };
         }
 
+        // We don't want to have memory leak so we need to unsubscribe any previous event if it exist.
+        // The unsubscribe must happen before SetParametersAsync.
+        if ( Nodes is INotifyCollectionChanged observableCollection1 )

```

Comment:
Maybe the collection variable name could be obseravableCollectionBeforeChange and observableColleftionAfterChange. Makes it a bit more clear if someone does. Not understand they might eb two different collections.